test/net/sourceforge/kolmafia/session/InventoryManagerTest.java

```diff
@@ -293,4 +299,29 @@ public void willNotRetrieveIfAlreadyBoughtFireworkHat() {
       }
     }
   }
+
+  @Nested
+  class PerUserVariableItems {
+    @Test
+    public void RingWillSetModsForUser() {
+      var builder = new FakeHttpClientBuilder();
+      var client = builder.client;
+
+      var cleanups = new Cleanups(withHttpClientBuilder(builder), withItem(ItemPool.RING, 1));
+      try (cleanups) {
+        String descId = ItemDatabase.getDescriptionId(ItemPool.RING);
+        var req = new GenericRequest("desc_item.php?whichitem=" + descId);
```

Comment:
You can prep the HttpClientBuilder to respond with certain strings to requests. Since you've set one up here, it probably makes more sense to use it rather than set req.responseText directly.

Reply:
sweet.  will update that after work.

Reply:
done